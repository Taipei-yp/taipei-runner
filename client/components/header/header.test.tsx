import React from "react";
import TestRenderer, { act } from "react-test-renderer";
import { Header } from ".";

describe("Component Header", () => {
  const leftChild = () => <div>leftChild</div>;
  const rightChild = () => <div>rightChild</div>;

  const header = TestRenderer.create(<Header className="testclass" />);

  it("correct props", () => {
    expect(header.root.props.className).toBe("testclass");
  });

  it("correct render without childs", () => {
    expect(header.toJSON()).toMatchSnapshot();
  });

  it("correct render with childs", () => {
    const headerWithChilds = TestRenderer.create(
      <Header className="testclass" left={leftChild()} right={rightChild()} />,
    );
    expect(headerWithChilds.toJSON()).toMatchSnapshot();
  });
});
