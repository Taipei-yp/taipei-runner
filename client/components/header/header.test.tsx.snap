// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Header correct render with childs 1`] = `
<div
  className={
    Object {
      "has": [Function],
      "is": [Function],
      "mix": [Function],
      "split": [Function],
      "state": [Function],
      "toString": [Function],
    }
  }
>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    <div>
      leftChild
    </div>
  </div>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    <div>
      rightChild
    </div>
  </div>
</div>
`;

exports[`Component Header correct render without childs 1`] = `
<div
  className={
    Object {
      "has": [Function],
      "is": [Function],
      "mix": [Function],
      "split": [Function],
      "state": [Function],
      "toString": [Function],
    }
  }
>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  />
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  />
</div>
`;
